lib/chef/resource/package.rb

```diff
@@ -31,14 +31,8 @@ class Package < Chef::Resource
 
       def initialize(name, run_context=nil)
         super
-        @candidate_version = nil
-        @options = nil
-        @package_name = name
-        @response_file = nil
+        package_name(name) unless name.nil?
         @response_file_variables = Hash.new
-        @source = nil
-        @version = nil
-        @timeout = nil
```

Comment:
One reason is to avoid implicit instance variable creation, which is a ruby warning (some day I would like Chef to run without warnings). The other is as pseudo documentation of which instance variables exist in a class.